src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3019,6 +3020,15 @@ SynthesizedRecordConstructor addCtor(RecordDeclarationSyntax declWithParameters)
                 return ctor;
             }
 
+            void addDeconstruct(ImmutableArray<ParameterSymbol> parameters, ImmutableArray<PropertySymbol> properties)
+            {
+                var ctor = new SynthesizedRecordDeconstruct(this, parameters, properties, memberOffset: members.Count);
+                if (!memberSignatures.ContainsKey(ctor))
```

Comment:
Looking at the comparer `DuplicateSourceComparer` which I think this uses, it seems that differences in ref kind are ignored. Can we test that? Is that what we want?

Reply:
Added the tests. I think it is fine to not synthesize `Deconstruct` for a 'ref'/'out' difference, for example. We happen to synthesize it for a 'None'/'out' difference, because it is allowed to overload on this difference, which also seems fine.